File: src/main/java/com/example/RDV/entities/DossierMedical.java
package com.example.RDV.entities;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;

@Entity
@Data
public class DossierMedical {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idDossier;
    private String etatClinique;
    private String groupe_sanguin;
    private String allergie;
    private String prescriptions_therapeutiques;
    private String  resultats_examen;
    private String observations;
    private String nomDuPatient;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private Date dateNaissancePatient;
    private Integer  cinPatient;
    private Integer telephonePatient;
    @JsonIgnore
    @OneToOne
    private Patient patient;

}

File: app/models/DossierMedical.ts
export interface DossierMedical {

        idDossier: number;
        etatClinique: string;
        groupe_sanguin: string;
        allergie: string;
        prescriptions_therapeutiques: string;
        resultats_examen: string;
        observations: string;
        nomDuPatient: string;
        dateNaissancePatient?: string;
        cinPatient: number;
        telephonePatient: number;
      }